src/Home.js

```diff
@@ -1,17 +1,87 @@
-import React from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import React, { useState, useEffect } from 'react';
+import { StyleSheet, Text, View, ScrollView, Switch } from 'react-native';
+import { DataStore } from 'aws-amplify';
+import { Todo } from './models';
 
 export default function Home() {
+
+    const [todos, setTodos] = useState([]);
+    let subscription;
+
+    useEffect(() => {
+        onQuery = async () => {
+            const todos_list = await DataStore.query(Todo);
+            setTodos(todos_list);
+        };
+
+        if (typeof subscription === 'undefined') {
+            subscription = DataStore.observe(Todo).subscribe((msg) => {
+                onQuery();
+            });
+        }
+
+        return function cleanup() {
+            subscription.unsubscribe();
+        }
+    }, []);
+
+    async function setComplete(updateValue, todo) {
+        await DataStore.save(
+            Todo.copyOf(todo, updated => {
+                updated.isComplete = updateValue
+            })
+        );
+    }
+
     return (
-        <View style={styles.container}>
-            <Text>Todo App</Text>
-        </View>
+        <ScrollView contentContainerStyle={styles.listContainer} >
+            {todos.map((todo, i) => (
+                <View key={i} style={styles.todoContainer}>
+                    <View>
+                        <Text style={styles.todoHeading} >{`${todo.name} `}</Text>
+                        <Text style={styles.todoDescription} >{`${todo.description} `}</Text>
+                    </View>
+                    <Switch
```

Comment:
Not fully sold on using a `Switch` in place of a `CheckBox`, the expected usage of a `Switch` in mobile UI is for system state